src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala

```diff
@@ -96,10 +96,18 @@ class AssetsRelation(config: RelationConfig)(val sqlContext: SQLContext)
   private val batchSize = config.batchSize.getOrElse(Constants.DefaultBatchSize)
 
   override def update(rows: Seq[Row]): IO[Unit] = {
-    val assetsItems = rows.map(r => fromRow[UpdateAssetsItemBase](r))
-    val updateAssetsItem = assetsItems.map(a => UpdateAssetsItem(a))
-
-    post(config, uri"${baseAssetsURL(config.project)}/update", updateAssetsItem)
+    val updateAssetsItems = rows
+      .map(r => fromRow[UpdateAssetsItemBase](r))
+      .map(a => UpdateAssetsItem(a))
+    updateAssetsItems
```

Comment:
Use [exists](https://www.scala-lang.org/api/2.13.x/scala/collection/Seq.html#exists(p:A=%3EBoolean):Boolean) and say `if (updateAssetsItems.exists(_.id.isEmpty))` to decide when to throw the exception.